app/auth.tsx

```diff
@@ -1,25 +1,38 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { Button, StyleSheet } from "react-native";
 import { ThemedView } from "@/components/ThemedView";
 import { ThemedText } from "@/components/ThemedText";
 import { useSession } from "@/hooks/ctx";
 import { router } from "expo-router";
+import { initDB, insertUser } from "./(tabs)/db-service";
 
 export default function AuthScreen() {
     const { signIn } = useSession();
 
+    // Initialize database when the app starts
+    useEffect(() => {
+        initDB();
+    }, []);
+
     const handleSignIn = async () => {
-        await signIn();
-        router.replace("/");
+        const userSession = await signIn(); // Ensure signIn() returns session data
+        if (userSession !== null) {
+            try {
+                const user = JSON.parse(userSession);
+                await insertUser(user.id, user.name, user.email, user.photo);
```

Comment:
This is going to get changed with my PR when I get information from the user so it will end up being moved